doc/function.md

```diff
@@ -0,0 +1,77 @@
+
```

Comment:
Move to /issues/lang/ For example, merge into https://github.com/functionalscript/functionalscript/blob/main/issues/lang/3110-function.md